src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Cmdlet/GetAzureSqlManagedInstance.cs

```diff
@@ -0,0 +1,128 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.ManagedInstance.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedInstance.Cmdlet
+{
+    /// <summary>
+    /// Defines the Get-ManagedInstance cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzureRmSqlManagedInstance", ConfirmImpact = ConfirmImpact.None)]
+    public class GetAzureSqlManagedInstance : ManagedInstanceCmdletBase
+    {
+        protected const string GetByNameAndResourceGroupParameterSet =
+            "GetManagedInstanceFromInputParameters";
+
+        protected const string GetByResourceIdParameterSet =
+            "GetManagedInstanceFromAzureResourceId";
+
+        /// <summary>
+        /// Gets or sets the name of the managed instance.
+        /// </summary>
+        [Parameter(ParameterSetName = GetByNameAndResourceGroupParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "SQL managed instance name.")]
+        [Alias("Name")]
+        [ValidateNotNullOrEmpty]
+        public string ManagedInstanceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the resource group.
+        /// </summary>
+        [Parameter(ParameterSetName = GetByNameAndResourceGroupParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
```

Comment:
Is it possible to give Name and not ResourceGroupName? If so, this is fine, but if not, you should have two parameter sets:

Get-AzureRmManagedInstance [-ResourceGroupName]
Get-AzureRmManagedInstance -ResourceGroupName -Name

Currently you are not taking care of the Name only scenario below.

Reply:
I see, I handled it in code better now. Invalid is only when Name is specified and ResourceGroup is not

Reply:
You need to fix the parameter sets so that this is obvious to users before running the cmdlet.  See the two parameter sets above.